protocol/x/revshare/keeper/revshare.go

```diff
@@ -142,3 +146,127 @@ func (k Keeper) ValidateRevShareSafety(
 	totalRevSharePpm := totalUnconditionalRevSharePpm + totalMarketMapperRevSharePpm + highestTierRevSharePpm
 	return totalRevSharePpm < lib.OneMillion
 }
+
+func (k Keeper) GetAllRevShares(
+	ctx sdk.Context,
+	fill clobtypes.FillForProcess,
+	monthlyRollingTakerVolumeQuantums uint64,
```

Comment:
@teddyding Added this extra field here instead of adding stakingkeeper to reduce dependencies. Also makes it easier to test

Reply:
Let's just add it to `FillForProcess`? Seems cleaner

Reply:
Also let's document that this rolling volume is the stats at the end of last block - as in, if trader has multiple fills in the block currently being processed, this trailing volume doesn't update intra block per fill.